exchanges/bittrex/bittrex_wrapper.go

```diff
@@ -711,12 +912,81 @@ func (b *Bittrex) ValidateCredentials(assetType asset.Item) error {
 	return b.CheckTransientError(err)
 }
 
+// FormatExchangeKlineInterval returns Interval to string
+// Overrides Base function
+func (b *Bittrex) FormatExchangeKlineInterval(in kline.Interval) string {
+	switch in {
+	case kline.OneMin:
+		return "MINUTE_1"
+	case kline.FiveMin:
+		return "MINUTE_5"
+	case kline.OneHour:
+		return "HOUR_1"
+	case kline.OneDay:
+		return "DAY_1"
+	default:
+		return "notfound"
+	}
+}
+
 // GetHistoricCandles returns candles between a time period for a set time interval
+// Candles set size returned by Bittrex depends on interval lenght:
+// - 1m interval: candles for 1 day (0:00 - 23:59)
+// - 5m interval: candles for 1 day (0:00 - 23:55)
+// - 1 hour interval: candles for 31 days
+// - 1 day interval: candles for 366 days
+// This implementation rounds returns candles up to the next interval or to the end
+// time (whichever comes first)
 func (b *Bittrex) GetHistoricCandles(pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	return kline.Item{}, common.ErrFunctionNotSupported
+	if err := b.ValidateKline(pair, a, interval); err != nil {
+		return kline.Item{}, err
+	}
+
+	formattedPair, err := b.FormatExchangeCurrency(pair, a)
+	if err != nil {
+		return kline.Item{}, err
+	}
+
+	year, month, day := start.Date()
+
+	ohlcData, err := b.GetHistoricalCandles(formattedPair.String(),
+		b.FormatExchangeKlineInterval(interval), year, int(month), day)
+	if err != nil {
+		return kline.Item{}, err
+	}
+
+	ret := kline.Item{
+		Exchange: b.Name,
+		Pair:     pair,
+		Asset:    a,
+		Interval: interval,
+	}
+
+	for x := range ohlcData {
+		timestamp, err := parseTime(ohlcData[x].StartsAt)
+		if err != nil {
+			return kline.Item{}, err
+		}
+		if timestamp.Before(start) {
+			continue
+		}
+		if timestamp.After(end) {
+			break
+		}
+		ret.Candles = append(ret.Candles, kline.Candle{
+			Time:   timestamp,
+			Open:   ohlcData[x].Open,
+			High:   ohlcData[x].High,
+			Low:    ohlcData[x].Low,
+			Close:  ohlcData[x].Close,
+			Volume: ohlcData[x].Volume,
+		})
+	}
+	return ret, nil
 }
 
 // GetHistoricCandlesExtended returns candles between a time period for a set time interval
 func (b *Bittrex) GetHistoricCandlesExtended(pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	return kline.Item{}, common.ErrFunctionNotSupported
+
```

Comment:
remove line

Reply:
I think the github diff gives a distored picture of what happened?